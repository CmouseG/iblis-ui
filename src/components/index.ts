export { //
    ChangePasswordForm, ChangePasswordFormProps, ChangePasswordFormValues, //
    DeleteAccountForm, DeleteAccountFormProps, DeleteAccountFormValues, //
    ManagedAddSimpleForm, ManagedAddSimpleFormProps, ManagedAddSimpleFormValues, //
    ManagedSimpleForm, ManagedSimpleFormProps, //
    ManagedSimpleListForm, ManagedSimpleListFormProps, ManagedSimpleListFormValues, //
    UserProfileForm, UserProfileFormProps, UserProfileFormValues
} from './form_components';
export {
    AccountCard, AccountCardProps, //
    AccountMenu, AccountMenuLevelProps, AccountMenuProps, //
    EmptyPage, //
    FilterButton, FilterButtonMenuProps, FilterButtonProps, //
    FooterMenu, FooterMenuLevelProps, FooterMenuProps, //
    IblisButton, IblisButtonProps, //
    IblisSelectField, IblisSelectFieldProps, //
    IblisSnackbar, IblisSnackbarProps, //
    IblisTextField, IblisTextFieldProps, //
    MainMenu, MainMenuLevelProps, MainMenuProps, //
    ProgressBar, ProgressBarProps, //
    UploadAvatar, UploadAvatarProps
} from './ui_components';
