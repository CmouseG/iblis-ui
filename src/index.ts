export {
    AccountCard, AccountCardProps, //
    AccountMenu, AccountMenuLevelProps, AccountMenuProps, //
    ChangePasswordForm, ChangePasswordFormProps, ChangePasswordFormValues, //
    ConfirmSignupForm, ConfirmSignupFormProps, ConfirmSignupFormValues, //
    DeleteAccountForm, DeleteAccountFormProps, DeleteAccountFormValues, //
    EmptyPage, //
    FilterButton, FilterButtonMenuProps, FilterButtonProps, //
    FooterMenu, FooterMenuLevelProps, FooterMenuProps, //
    ForceChangePasswordForm, ForceChangePasswordFormProps, ForceChangePasswordFormValues, //
    HelperButton, HelperButtonProps, //
    IblisButton, IblisButtonProps, //
    IblisSelectField, IblisSelectFieldProps, //
    IblisSnackbar, IblisSnackbarProps, IblisTextField, IblisTextFieldProps, //
    LoginForm, LoginFormProps, LoginFormValues, //
    MainMenu, MainMenuLevelProps, MainMenuProps, //
    ManagedAddSimpleForm, ManagedAddSimpleFormProps, ManagedAddSimpleFormValues, //
    ManagedSimpleForm, ManagedSimpleFormData, ManagedSimpleFormProps, //
    ManagedSimpleListForm, ManagedSimpleListFormProps, ManagedSimpleListFormValues, //
    PasswordConfirmForm, PasswordConfirmFormProps, PasswordConfirmFormValues, //
    ProgressBar, ProgressBarProps, //
    ResetPasswordForm, ResetPasswordFormProps, ResetPasswordFormValues, //
    SignupForm, SignupFormProps, SignupFormValues, //
    StripeForm, StripeFormProps, //
    UploadAvatar, UploadAvatarProps, //
    UserProfileForm, UserProfileFormProps, UserProfileFormValues //
} from './components';
export { //
    CardMenu, CardMenuLevelProps, CardMenuProps, //
    CardTitleLayout, CardTitleLayoutProps, //
    ContentWithTitle, ContentWithTitleProps, //
    ExternalFormLayout, ExternalFormLayoutProps, //
    MainLayout, MainLayoutProps
} from './layouts';
export { theme } from './theme';
// tslint:disable-next-line:no-import-side-effect
import 'iblis-font/index.css';
// tslint:disable-next-line:no-import-side-effect
import 'typeface-roboto/index.css';
