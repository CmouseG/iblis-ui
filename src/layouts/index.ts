export { CardMenu, CardMenuLevelProps, CardMenuProps } from './card_menu';
export { CardTitleLayout, CardTitleLayoutProps } from './card_title_layout';
export { ContentWithTitle, ContentWithTitleProps } from './content_with_title';
export { ExternalFormLayout, ExternalFormLayoutProps } from './external_form_layout';
export { MainLayout, MainLayoutProps } from './main_layout';
