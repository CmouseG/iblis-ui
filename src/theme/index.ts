export { theme } from './iblis_theme';
