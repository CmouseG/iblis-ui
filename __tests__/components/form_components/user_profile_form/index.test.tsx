import * as React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { combineReducers, createStore, ReducersMapObject, Store } from 'redux';
import { ConfigProps, reducer as formReducer, reduxForm } from 'redux-form';
import { UserProfileForm, UserProfileFormProps, UserProfileFormValues } from '../../../../src';

it('renders correctly', () => {

    // Combine all app Reducers
    const reducersList: ReducersMapObject = {
        form: formReducer,
    };

    // Combine all app Reducers
    const reducers = combineReducers(reducersList);

    // create redux store with all app reducers amd all the middleware
    const store: Store<{}> = createStore(
        // reducers
        reducers,
        // preload state
        {},
    );

    const langValues = [{ id: 'en', value: 'English' }, { id: 'fr', value: 'French' }];

    // configure the form
    function config(): ConfigProps<UserProfileFormValues, UserProfileFormProps> {
        const configForm: ConfigProps<UserProfileFormValues, UserProfileFormProps> = {
            form: 'form',
        };
        return configForm;
    }
    // connect the form
    const ConnectedForm = reduxForm(config())(UserProfileForm);
    const userProfileFormValues: UserProfileFormValues = {
        firstName: 'Albert',
        lastName: 'Einstein',
        email: 'albert@einstein.com',
        language: 'en',
        avatarFileId: '22',
    };

    const tree = renderer
        .create(
            <Provider store={store}>
                <ConnectedForm
                    initValues={userProfileFormValues}
                    firstNameLabel={'FirstName'}
                    lastNameLabel={'LastName'}
                    emailLabel={'Email'}
                    languageLabel={'Language'}
                    buttonLabelConfirm={'Update'}
                    requiredErrorLabel={'Required'}
                    emailNotValidErrorLabel={'Email not valid'}
                    languagePossibleValues={langValues}
                    avatarComponent={<div />}
                    isLoading={false}
                    updateUser={(_values) => undefined}
                />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
