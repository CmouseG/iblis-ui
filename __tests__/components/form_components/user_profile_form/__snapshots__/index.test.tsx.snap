// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-2-44"
    >
      <div />
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-1-43"
    >
       
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-9-51"
    >
      <form
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiGrid-container-1"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-2-44"
          >
            <aside
              className="MuiTypography-root-97 MuiTypography-body2-105"
            >
              FirstName
            </aside>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-48"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <div
                  className="MuiFormControl-root-123 IblisTextFieldBase-textFieldRoot-121 MuiFormControl-fullWidth-126"
                >
                  <div
                    className="MuiInput-root-127 MuiInput-fullWidth-134 MuiInput-formControl-128"
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInput-input-135 IblisTextFieldBase-textFieldInput-122"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                      type="text"
                      value="Albert"
                    />
                  </div>
                </div>
                <span
                  className="MuiTypography-root-97 MuiTypography-caption-107 MuiTypography-colorError-120"
                >
                  *
                </span>
              </div>
              <span
                className="MuiTypography-root-97 MuiTypography-caption-107 MuiTypography-colorError-120"
              >
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-container-1"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-2-44"
          >
            <aside
              className="MuiTypography-root-97 MuiTypography-body2-105"
            >
              LastName
            </aside>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-48"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <div
                  className="MuiFormControl-root-123 IblisTextFieldBase-textFieldRoot-121 MuiFormControl-fullWidth-126"
                >
                  <div
                    className="MuiInput-root-127 MuiInput-fullWidth-134 MuiInput-formControl-128"
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInput-input-135 IblisTextFieldBase-textFieldInput-122"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                      type="text"
                      value="Einstein"
                    />
                  </div>
                </div>
                <span
                  className="MuiTypography-root-97 MuiTypography-caption-107 MuiTypography-colorError-120"
                >
                  *
                </span>
              </div>
              <span
                className="MuiTypography-root-97 MuiTypography-caption-107 MuiTypography-colorError-120"
              >
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-container-1"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-2-44"
          >
            <aside
              className="MuiTypography-root-97 MuiTypography-body2-105"
            >
              Email
            </aside>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-48"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <div
                  className="MuiFormControl-root-123 IblisTextFieldBase-textFieldRoot-121 MuiFormControl-fullWidth-126"
                >
                  <div
                    className="MuiInput-root-127 MuiInput-fullWidth-134 MuiInput-formControl-128"
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInput-input-135 IblisTextFieldBase-textFieldInput-122"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={true}
                      type="text"
                      value="albert@einstein.com"
                    />
                  </div>
                </div>
                <span
                  className="MuiTypography-root-97 MuiTypography-caption-107 MuiTypography-colorError-120"
                >
                  *
                </span>
              </div>
              <span
                className="MuiTypography-root-97 MuiTypography-caption-107 MuiTypography-colorError-120"
              >
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-container-1"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-2-44"
          >
            <aside
              className="MuiTypography-root-97 MuiTypography-body2-105"
            >
              Language
            </aside>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-48"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <div
                  className="MuiFormControl-root-123 IblisSelectFieldBase-textFieldRoot-140 MuiFormControl-fullWidth-126"
                >
                  <div
                    className="MuiInput-root-127 MuiInput-fullWidth-134 MuiInput-formControl-128"
                  >
                    <div
                      className="MuiSelect-root-143"
                    >
                      <div
                        aria-haspopup="true"
                        aria-owns={null}
                        aria-pressed="false"
                        className="MuiSelect-select-144 MuiSelect-selectMenu-145 MuiInput-input-135 IblisSelectFieldBase-textFieldInput-141 MuiInput-inputType-138"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        English
                      </div>
                      <input
                        autoFocus={false}
                        type="hidden"
                        value="en"
                      />
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-148 MuiSelect-icon-147 IblisSelectFieldBase-iconSelectTextFieldInput-142"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <span
                  className="MuiTypography-root-97 MuiTypography-caption-107 MuiTypography-colorError-120"
                >
                  *
                </span>
              </div>
              <span
                className="MuiTypography-root-97 MuiTypography-caption-107 MuiTypography-colorError-120"
              >
                 
              </span>
            </div>
          </div>
        </div>
        <div>
          <div
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <div
              className="MuiFormControl-root-123 IblisTextFieldBase-textFieldRoot-121"
            >
              <div
                className="MuiInput-root-127 MuiInput-formControl-128"
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInput-input-135 IblisTextFieldBase-textFieldInput-122 MuiInput-inputType-138"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="hidden"
                  value="22"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-container-1"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-sm-2-44"
          >
             
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-sm-6-48"
          >
            <button
              className="MuiButtonBase-root-183 MuiButton-root-159 MuiButton-contained-168 MuiButton-containedPrimary-169 MuiButton-raised-171 MuiButton-raisedPrimary-172"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="submit"
            >
              <span
                className="MuiButton-label-160"
              >
                Update
              </span>
              <span
                className="MuiTouchRipple-root-186"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
